GDJS/Runtime/inputmanager.js

```diff
@@ -41,22 +41,52 @@ gdjs.InputManager.MOUSE_RIGHT_BUTTON = 1;
 /** @constant {number} */
 gdjs.InputManager.MOUSE_MIDDLE_BUTTON = 2;
 
+/**
+ * Returns the location-modified keyCode.
+ *
+ * @private
+ */
+gdjs.InputManager.prototype.getModifiedKeyCode = function(keyCode, location) {
+  if (keyCode > 0 && location) {
+    // If it is a numpad number, do not modify it.
+    if (96 <= keyCode && keyCode <= 105) {
+      return keyCode;
+    }
+
+    if (
+      location === KeyboardEvent.DOM_KEY_LOCATION_RIGHT ||
+      location === KeyboardEvent.DOM_KEY_LOCATION_NUMPAD
+    ) {
+      return keyCode + 1000;
```

Comment:
What do you think of `keyCode + 1000 * location`?

So that:
standard key = usual key code
left key = keycode + 1000
right key = keycode + 2000
numpad (with the exception of numpad 0-9) = keycode + 3000

This seems an interesting encoding because we can deduce the location from the resulting keycode.

Reply:
To finalize this and be compatible with libraries that are not able to give us the location (i.e: Cocos2d-JS), we'll need to add special casing to handle these (much like you already did for the numpad number, which are not consistent):

If location is undefined, Shift (16), Ctrl (17), Alt (18) and System (91) should be converted to their left variant (this is a choice we're making) OR we choose to simulate both keys to be pressed, in which case, in `onKeyPressed`/`onKeyReleased`, we handle this special case of location being undefined.

Up to you! Might be simpler to just convert them to their left variant.

Reply:
Just fixed it - would you have a look?